[TensorExpr] IRVerifier: add index verifier for Store.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53137 [TensorExpr] IRVerifier: add index verifier for Store.**

Also, add casting to Int for Load and Store indices.

Fixes #52773.

Differential Revision: [D26760256](https://our.internmc.facebook.com/intern/diff/D26760256)

Commit:
[TensorExpr] IRVerifier: add index verifier for Store.

Also, add casting to Int for Load and Store indices.

Fixes #52773.

[ghstack-poisoned]

navahgar commented on torch/csrc/jit/tensorexpr/ir.cpp:
```diff
@@ -23,7 +23,17 @@ Load::Load(
+    if (is_integral(dt.scalar_type()) && dt.scalar_type() != index_dtype) {
+      indices_[i] = new Cast(Dtype(index_dtype, dt.lanes()), indices_[i]);
+    }
+  }
```
Extract this into a util function and reuse below?

navahgar reviewed (approved):
LGTM.

Just one minor nit.

Commit:
Update on "[TensorExpr] IRVerifier: add index verifier for Store."


Also, add casting to Int for Load and Store indices.

Fixes #52773.

Differential Revision: [D26760256](https://our.internmc.facebook.com/intern/diff/D26760256)

[ghstack-poisoned]
